app/Http/Middleware/RoomAuthenticate.php

```diff
@@ -27,7 +28,7 @@ public function handle($request, Closure $next, $allowAuthenticated = false)
         $room          = $request->route('room');
 
         // requested user is the owner or a member of the room or the room doesn't require access code
-        if ($room->owner->is(Auth::user()) or $room->members->contains(Auth::user()) or $room->accessCode == null) {
+        if ($room->accessCode == null || (Auth::user() && ($room->owner->is(Auth::user()) || $room->members->contains(Auth::user()) || Auth::user()->can('viewAll', Room::class)))) {
```

Comment:
Is this extra `Auth::user()` check necessary?

Reply:
Yes, if the accessCode is not null and the request comes from a guest Auth::user() would be null. Checking Auth::user()->can('viewAll', Room::class) would therefore result in an error.

Reply:
But then the check is at a wrong place!